Sources/StreamChatUI/Components.swift

```diff
@@ -100,6 +100,12 @@ public struct Components {
     /// Used in both the Channel and Thread view controllers.
     @available(iOSApplicationExtension, unavailable)
     public var messageListVC: ChatMessageListVC.Type = ChatMessageListVC.self
+
+    /// A closure to bypass the message list channel data source.
+    public var messagesBypass: (([ChatMessage]) -> [ChatMessage])?
```

Comment:
One thing that I would like to point out is that blocks are more difficult to debug and provide useful information in stacktraces (after a crash). I would be more confident with a new protocol(maybe MessageFiltering) & object(provided from clients) pair.